xdsl/backend/x86/lowering/convert_ptr_to_x86.py

```diff
@@ -0,0 +1,92 @@
+from dataclasses import dataclass
+from functools import reduce
+from typing import cast
+
+from xdsl.context import Context
+from xdsl.dialects import builtin, ptr, x86
+from xdsl.dialects.builtin import (
+    ArrayAttr,
+    FixedBitwidthType,
+    IntAttr,
+    UnrealizedConversionCastOp,
+    VectorType,
+)
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.utils.exceptions import DiagnosticException
+
+
+@dataclass
+class PtrLoadToX86(RewritePattern):
+    arch: str
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: ptr.LoadOp, rewriter: PatternRewriter):
+        source = op.addr
+        value = op.res
+        value_type = value.type
+        if isinstance(value_type, VectorType):
+            # Pointer cast
+            x86_reg_type = x86.register.UNALLOCATED_GENERAL
+            cast_op = UnrealizedConversionCastOp.get((source,), (x86_reg_type,))
+            # Output vector description
+            vector_int_shape = [i.data for i in value_type.shape.data]
+            vector_shape = ArrayAttr(
+                [IntAttr(reduce(lambda x, y: x * y, vector_int_shape))]
+            )
+            vector_num_elements = vector_shape.data[0].data
+            element_type = cast(FixedBitwidthType, value_type.get_element_type())
```

Comment:
Two proposals: use the helper on ShapedType to get the number of elements, and to try and avoid the cast, it's a bit of a black hole for the type checker, I think paying the cost of an isinstance check is worth it for things like this.
```suggestion
            vector_num_elements = value_type.element_count()
            element_type = value_type.get_element_type()
            assert isinstance(element_type, FixedBitwidthType)
```

Reply:
Ok, done for the helper. However, replacing ```cast``` with ```assert isinstance``` here does not eliminate the type error. So, I propose to keep ```cast```.